emr-user-role-mapper-s3storagebasedauthorizationmanager/pom.xml

```diff
@@ -22,23 +26,23 @@
         <dependency>
             <groupId>org.apache.hive</groupId>
             <artifactId>hive-common</artifactId>
-            <version>3.1.2</version>
+            <version>${hive.version}</version>
```

Comment:
I have a concern here, Hive 2 and Hive 3, this plugin will have a code incompatibility issue. Specifically at this function: public void setMetaStoreHandler(IHMSHandler handler). In Hive 3 it is an interface, but in Hive 2 it is a abstract class HMSHandler. So it seems we have to version branch this to make it both work for hive 2 and 3.

Reply:
We already know that this limitation exists. We have documented in the README file:

"Change *hive.version* in pom.xml for the version of hive that you will be using. It should work with Hive 3.x, however, if Hive 2.x is needed, then some code changes may be necessary. "